libnestutil/type_traits.h

```diff
@@ -71,7 +71,7 @@ struct is_floating_point;
 template < class T >
 struct is_pointer;
 // MSVC can't compile this correctly, and neither can gcc 3.3.5 (at least)
-#if !defined( _MSC_VER ) && !( defined( __GNUC__ ) && __GNUC__ <= 3 )
+#if not defined( _MSC_VER ) && not( defined( __GNUC__ ) && __GNUC__ <= 3 )
```

Comment:
'not' isn't supported by the preprocessor language. Since 'not' is syntactically correct (it is a valid macro name), the preprocessor will generate a warning in the best case. 'not' should be replaced with '!' or the expression can also be re-written using #ifndef.